src/components/timepicker/timepicker.tsx

```diff
@@ -0,0 +1,216 @@
+moment.updateLocale("en", i18n.en);
+moment.updateLocale("de", i18n.de);
+
+interface CalendarCell {
+  date: moment.Moment;
+  label: string;
+  dateString: string;
+  isToday: boolean;
+  isSelected: boolean;
+  isDisabled: boolean;
+  isOutdated: boolean;
+}
+
+export type TimeCallback = (date: string) => boolean;
+
+@Component({
+  tag: "bal-timepicker",
+  styleUrl: "timepicker.scss",
+})
+export class Timepicker {
+  static FORMAT = "HH:mm";
+
+  dropDownElement!: HTMLBalDropdownElement;
+  now: moment.Moment = moment().startOf("hour");
+
+  @State() isPristine = true;
+  @State() selectedTime: moment.Moment = moment().startOf("hour");
+  @State() pointerTime: moment.Moment = moment().startOf("hour");
+
+  /**
+   * Language of the timepicker. Possible values are `de`, `fr`,`it` or `en`.
+   */
+  @Prop() language: string = "de";
+
+  /**
+   * Placeholder text to render if no time has been selected.
+   */
+  @Prop() placeholder: string = "Click to select...";
+
+  /**
+   * Disable the input
+   */
+  @Prop() disabled: boolean = false;
+
+  /**
+   * Latest time available for selection
+   */
+  @Prop() maxTime: string = "";
+
+  /**
+   * Earliest time available for selection
+   */
+  @Prop() minTime: string = "";
+
+  /**
+   * The value of the timepicker with the format `HH:mm`.
+   */
+  @Prop({ mutable: true }) value: string = "";
+
+  @Watch("value")
+  valueWatcher(newValue: string) {
+    this.isPristine = false;
+    this.selectedTime = moment(newValue, Timepicker.FORMAT);
+    this.pointerTime = moment(newValue, Timepicker.FORMAT);
+  }
+
+  /**
+   * Callback to determine which time in the timepicker should be selectable.
+   */
+  @Prop() filter: TimeCallback = (_) => true;
+
+  /**
+   * Triggers when the value of the timepicker is changed
+   */
+  @Event({
+    eventName: "balChange",
+    composed: true,
+    cancelable: true,
+    bubbles: true,
+  }) balChangeEventEmitter!: EventEmitter<string>;
+
+  componentWillLoad() {
+    moment.locale(this.language);
+    if (this.value) {
+      this.isPristine = false;
+      this.selectedTime = moment(this.value, Timepicker.FORMAT);
+      this.pointerTime = moment(this.value, Timepicker.FORMAT);
+    }
+  }
+
+  /**
+   * Open the timepicker dropdown
+   */
+  @Method()
+  async open() {
+    await this.dropDownElement.open();
+  }
+
+  /**
+   * Close the timepicker dropdown
+   */
+  @Method()
+  async close() {
+    await this.dropDownElement.close();
+  }
+
+  get weekdays(): string[] {
+    const arr: string[] = moment.weekdaysMin();
+    arr.push(arr.shift() as any);
+    return arr;
+  }
+
+  get parsedMaxTime(): moment.Moment {
+    return moment(this.maxTime, Timepicker.FORMAT).startOf("hour");
+  }
+
+  get parsedMinTime(): moment.Moment {
+    return moment(this.minTime, Timepicker.FORMAT).startOf("hour");
+  }
+
+  get firstDateOfBox(): moment.Moment {
+    return moment(this.pointerTime).startOf("day");
+  }
+
+  get calendarGrid(): CalendarCell[][] {
```

Comment:
Could you generate the time grid without moment?
This would be much faster ;-)

Reply:
The more important question would be: Do we need moment at all? At first glance it makes sense to drop it entirely but then it would be harder to extend the timepicker in the future to support time slots like "12:15", "12:30" etc. But on the other hand, such ability would require a redesign of the matrix, as the number of elements would become to large (24*4 if you want set a time every quarter of an hour). My suggestion is to leave it for now and discuss it in a new issue.